File: socialNetwork/src/main/java/com/ftn/socialNetwork/model/entity/Report.java
package com.ftn.socialNetwork.model.entity;

import javax.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "reports")
public class Report {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(nullable = false)
    private EReportReason reason;
    @Column(nullable = false, columnDefinition = "DATETIME DEFAULT CURRENT_TIMESTAMP")
    private LocalDate timestamp;
    @Column(nullable = false)
    private boolean accepted;
  @Column(nullable = false, columnDefinition = "boolean default false")
  private Boolean isDeleted ;
    @ManyToOne
    private User user;
    @ManyToOne
    private Post post;
    @ManyToOne
    private Comment comment;
    @ManyToOne
    private User reportedUser;
}

File: src/app/models/report.ts
import { User } from "./user";
import { Post } from "./post";
import { EReportReason } from "./eReportReason";
import { Comment } from "./comment";

export class Report {
  id: number;
  reason: EReportReason;
  timestamp: Date;
  accepted: boolean;
  isDeleted: boolean;
  user: User  | undefined;
  post: Post | undefined;
  comment: Comment | undefined;
  reportedUser: User | undefined; 

  constructor(
    id: number,
    reason: EReportReason,
    timestamp: Date,
    accepted: boolean,
    isDeleted: boolean,
    user: User,
    post: Post,
    comment: Comment,
    reportedUser: User
  ) {
    this.id = id;
    this.reason = reason;
    this.timestamp = timestamp;
    this.accepted = accepted;
    this.isDeleted = isDeleted;
    this.user = user;
    this.post = post;
    this.comment = comment;
    this.reportedUser = reportedUser;
  }
}
